packages/flutter_tools/test/commands/run_test.dart

```diff
@@ -28,6 +43,75 @@ void main() {
       }
     });
 
+    testUsingContext('dart-flags option is not available on stable channel', () async {
+      when(mockDeviceManager.getDevices()).thenAnswer((Invocation invocation) {
+        return Stream<Device>.fromIterable(<Device>[
+          FakeDevice(),
+        ]);
+      });
+
+      final RunCommand command = TestRunCommand();
+      final List<String> args = <String> [
+                                  'run',
+                                  '--dart-flags', '"--observe"',
+                                  '--no-hot',
+                                ];
+
+      // Stable branch.
+      try {
+        await createTestCommandRunner(command).run(args);
+        fail('Expect exception');
+      // ignore: unused_catch_clause
+      } on UsageException catch(e) {
+        // Not available while on stable branch.
+      }
+    }, overrides: <Type, Generator>{
+      DeviceManager: () => mockDeviceManager,
+      FlutterVersion: () => mockStableFlutterVersion,
+    });
+
+    testUsingContext('dart-flags option is only populated for debug and profile modes', () async {
+      when(mockDeviceManager.getDevices()).thenAnswer((Invocation invocation) {
+        return Stream<Device>.fromIterable(<Device>[
+          FakeDevice(),
+        ]);
+      });
+      final RunCommand command = TestRunCommand();
+      final List<String> args = <String> [
+                                  'run',
+                                  '--dart-flags', '"--observe"',
+                                  '--no-hot',
+                                ];
+
+      // Debug mode
+      try {
+        await createTestCommandRunner(command).run(args);
```

Comment:
I'm not sure what these cases are catching, we're not checking the argResults for the flags so how do we know the dart-flags is/isn't accepted?

Reply:
Testing of which flags are actually accepted is done in the engine since that's where the whitelist lives. This is just making sure that dart-flags isn't accepted while on the stable branch or in release mode.